hrms/hr/doctype/salary_withholding/salary_withholding.py

```diff
@@ -0,0 +1,48 @@
+# Copyright (c) 2024, Frappe Technologies Pvt. Ltd. and contributors
+# For license information, please see license.txt
+
+import frappe
+from frappe.model.document import Document
+
+
+class SalaryWithholding(Document):
+	pass
+
+
+@frappe.whitelist()
+def get_employee_details(employee):
+	salary_structure_assignment = frappe.db.get_value(
+		"Salary Structure Assignment",
+		{"employee": employee},
+		"salary_structure",
+	)
+
+	payroll_frequency = frappe.db.get_value(
+		"Salary Structure", salary_structure_assignment, "payroll_frequency"
+	)
+
+	details = {
+		"date_of_joining": frappe.get_value("Employee", employee, "date_of_joining"),
+		"relieving_date": frappe.get_value("Employee", employee, "relieving_date"),
+		"employee_name": frappe.get_value("Employee", employee, "employee_name"),
+		"resignation_letter_date": frappe.get_value("Employee", employee, "resignation_letter_date"),
+		"notice_number_of_days": frappe.get_value("Employee", employee, "notice_number_of_days"),
```

Comment:
This will lead to 5 DB calls from a single table. All of this should be fetched in a single query. You can just set fetch from properties for these fields and eliminate this code completely